### PR TITLE
MVP

### PR DESCRIPTION
**Pull request by Corniel**
Test tools that allow to define expected behavior of Roslyn Code Analyzers with annotated code files; inspired by, but not dependent on SonarSource.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+[SonarSource](https://sonarsource.com). The same patterns are used, and their
+[implementation](https://github.com/SonarSource/sonar-dotnet) has been an
+inspiration for this one.
+
+## Defining expected issues
+The basic idea is that by adding comments (of a certain format) to a code file
+the verifier compare the actual raised issues and errors to the expected onces.
+If there are differences, the verification fails.
+
+The easiest way to this, is adding `// Noncompliant` to the end of a line
```
Comment (mars888) [inline README.md:16]: `to this..` => `to _do_ this, is _by_ adding`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+that contains an issue:
+``` C#
+class MyClass { } // Noncompliant
+```
+
+By adding also `{{Some expected message}}` the message itself is also checked:
```
Comment (mars888) [inline README.md:22]: => `By also adding`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+### Expected errors
+Instead of a line of code being noncompliant, it is possible to mark a line as
+an error too. Just use `Error` instead of `Noncompliant`:
+
+``` C#
+public class Missing { // Error [CS1513]
+```
+
+## Setup a verification test
+A test can be setup up justing a test framework of choice. The examples here
```
Comment (mars888) [inline README.md:74]: => `setup ~~up~~`

Comment (mars888) [reply README.md:74]: `justing` => `using`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+[Test]
+public void Verify_MyVBAnalyzer()
+    => new Verify_MyVBAnalyzer()
+        .ForVB()
+        .AddSource("Path\to\testcode.vb")
+        .Verify();
+```
+
+This will run your analyzer on the provide code, and throw a
```
Comment (mars888) [inline README.md:94]: => `provided`

File: README.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+The code base is compiled against the latest language version (currently C# 9,
+and Visual Basic 16.9).
+
+### Tweaking the Verify Analyzer Context
+The initialized context has been for one language. All changes/additions take
+that in to consideration. So extra analyzers/source files, etcetera, must be
+supported by that language.
+
+### Additional analyzers
+Extra analyzers can be added using `.Add(iagnosticAnalyzer analyzer)`.
```
Comment (mars888) [inline README.md:114]: => `DiagnosticAnalyzer`?

**Comment by Corniel** [comment]
@mars888 Applied all suggestions.

**Pull request merged**